frontend/src/models/phase.ts

```diff
@@ -0,0 +1,35 @@
+enum PHASE_NAMES {
+  PHASE_0 = "phase 0",
+  PHASE_1 = "phase 1",
+  PHASE_2 = "phase 2",
+  SANDBOX = "sandbox",
+}
+
+// phase data for the frontend
+class Phase {
+  constructor(
+    name: PHASE_NAMES,
+    index: number,
+    info: string,
+    preamble: string,
+    path_to_documents: string,
+    isComplete: boolean,
+    isCurrent: boolean
+  ) {
+    this.name = name;
+    this.index = index;
+    this.info = info;
+    this.preamble = preamble;
+    this.path_to_documents = path_to_documents;
+    this.isCurrent = isCurrent;
+    this.isComplete = isComplete;
+  }
+  name: PHASE_NAMES;
+  index: number;
+  info: string;
+  preamble: string;
+  path_to_documents: string;
+  isCurrent: boolean;
+  isComplete: boolean;
```

Comment:
The frontend won't be bothered about some of these fields, I'd change it to something like 
```
{
  id: PHASE_NAMES; // unique identifier
  name: string; // display name
  preamble: string;
  isCurrent: boolean;
  isComplete: boolean;
}
```